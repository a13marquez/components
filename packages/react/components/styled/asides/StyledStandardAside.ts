import styled from 'styled-components';

export const StyledStandardAside = styled.aside`
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
    font: 100%;
    display: block;
`;