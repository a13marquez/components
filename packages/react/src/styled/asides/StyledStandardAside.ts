import styled from 'styled-components';

const StyledStandardAside = styled.aside`
    display: block;
    margin: 0;
    padding: 0;
    border: 0;
    font: 100%;
`;

export default StyledStandardAside;