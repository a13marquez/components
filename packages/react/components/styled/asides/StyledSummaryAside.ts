import styled from 'styled-components';
import { StyledStandardAside } from './StyledStandardAside';

export const StyledSummaryAside = styled(StyledStandardAside)`
  display: flex;
  flex-flow: column wrap;
  padding: 40px 32px;
  width: 312px;
  background-color: #f3f3f3;
  color: #212240;
  > * {
    width: 100%;
  }
`