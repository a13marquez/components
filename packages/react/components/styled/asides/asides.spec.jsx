import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import { StyledStandardAside } from './StyledStandardAside';
import { StyledSummaryAside } from './StyledSummaryAside';


test('It renders Styled Standard Aside', () => {
  const StandardAside = mount(<StyledStandardAside />);
  expect(StandardAside).toMatchSnapshot();
});

test('It renders Styled Summary Aside', () => {
  const SummaryAside = mount(<StyledSummaryAside />);
  expect(SummaryAside).toMatchSnapshot();
});

test('Styled Standard Aside have the correct styles', () => {
  const StandardAside = mount(<StyledStandardAside />);
  expect(StandardAside).toHaveStyleRule('margin', '0');
  expect(StandardAside).toHaveStyleRule('padding', '0');
  expect(StandardAside).toHaveStyleRule('border', '0');
  expect(StandardAside).toHaveStyleRule('vertical-align', 'baseline');
  expect(StandardAside).toHaveStyleRule('font', '100%');
  expect(StandardAside).toHaveStyleRule('display', 'block');
});


test('Styled Summary Aside have the correct styles', () => {
  const SummaryAside = mount(<StyledSummaryAside />);
  expect(SummaryAside).toHaveStyleRule('margin', '0');
  expect(SummaryAside).toHaveStyleRule('border', '0');
  expect(SummaryAside).toHaveStyleRule('vertical-align', 'baseline');
  expect(SummaryAside).toHaveStyleRule('font', '100%');
  expect(SummaryAside).toHaveStyleRule('display', 'flex');
  expect(SummaryAside).toHaveStyleRule('flex-flow', 'column wrap');
  expect(SummaryAside).toHaveStyleRule('padding', '40px 32px');
  expect(SummaryAside).toHaveStyleRule('width', '312px');
  expect(SummaryAside).toHaveStyleRule('background-color', '#f3f3f3');
  expect(SummaryAside).toHaveStyleRule('color', '#212240');
});
