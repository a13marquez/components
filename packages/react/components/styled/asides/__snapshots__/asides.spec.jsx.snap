// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders Styled Standard Aside 1`] = `ReactWrapper {}`;

exports[`It renders Styled Summary Aside 1`] = `ReactWrapper {}`;
